plugins/kubernetes/app/models/kubernetes/build_job_executor.rb

```diff
@@ -0,0 +1,140 @@
+# Execute a job that can build/push Docker images
+# and write progress to the local job output.
+module Kubernetes
+  class BuildJobExecutor
+    WAIT_FOR_JOB = 5.minutes
+    TICK = 3.seconds
+
+    attr_reader :job
+    delegate :id, to: :job
+
+    def initialize(output, job:)
+      @output = output
+      @job = job
+      @registry = {}
+    end
+
+    def execute!(build, project, tag:, push: false, registry:, tag_as_latest: false)
+      job_log = job_name = job_namespace = ""
```

Comment:
Just my habit to declare all variables before using them to deal with code coverage or normal static analysis tool. Is this a bad thing in Ruby/Rails?